File: c/challenge_121_easy.c
#include <stdio.h>

int main(int argc, char *argv[], char **envp) {
  int coin = atoi(argv[1]);
  int result = run(coin);
  printf("%d\n", result);
}

int run(int coin) {
  if(coin == 0) {
    return 1;
  }

  return run(coin / 2) +
         run(coin / 3) +
         run(coin / 4);
}

File: js/challenge_121_easy.js

var coin = process.argv[2];

console.log(run(parseInt(coin)));

function run(coin) {
  if(coin === 0) return 1;
  else return run(Math.floor(coin / 2)) + 
              run(Math.floor(coin / 3)) + 
              run(Math.floor(coin / 4));
}

